contributions/api/controllers/contributions.py

```diff
@@ -0,0 +1,574 @@
+#  Copyright 2020 Board of Trustees of the University of Illinois.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import datetime
+import logging
+import copy
+
+from flask import jsonify, request, g
+from bson import ObjectId
+
+import controllers.configs as cfg
+import utils.jsonutils as jsonutils
+import utils.datasetutils as datasetutils
+import utils.rest_handlers as rs_handlers
+import utils.mongoutils as mongoutils
+
+from utils import query_params
+from models.contribution import Contribution
+from models.person import Person
+from models.organization import Organization
+from models.capabilities.capability import Capability
+from models.talents.talent import Talent
+from pymongo import MongoClient
+
+client_contribution = MongoClient(cfg.MONGO_CONTRIBUTION_URL, connect=False)
+db_contribution = client_contribution[cfg.CONTRIBUTION_DB_NAME]
+coll_contribution = db_contribution[cfg.CONTRIBUTION_COLL_NAME]
+
+def post():
+    is_new_install = True
+
+    # check if name is in there otherwise it is either a first installation
+    try:
+        in_json = request.get_json()
+        name = in_json[cfg.FIELD_NAME]
+        # check if the dataset is existing with given name
+        dataset = mongoutils.get_contribution_dataset_from_field(coll_contribution, cfg.FIELD_NAME, name)
+        if dataset is not None:
+            is_new_install = False
+            msg = {
+                "reason": "NAME in input json already exists in the database: " + str(name),
+                "error": "Bad Request: " + request.url,
+            }
+            msg_json = jsonutils.create_log_json("Contribution", "POST", msg)
+            logging.error("POST " + json.dumps(msg_json))
+            return rs_handlers.bad_request(msg)
+    except:
+        pass
+
+    if is_new_install:
+        # new installation of the app
+        currenttime = datetime.datetime.now()
+        currenttime = currenttime.strftime("%Y/%m/%dT%H:%M:%S")
+        contribution_dataset = Contribution('')
+        contribution_dataset, restjson = datasetutils.update_capability_dataset_from_json(contribution_dataset, in_json)
```

Comment:
```suggestion
        contribution_dataset, restjson = datasetutils.update_contribution_dataset_from_json(contribution_dataset, in_json)
```
I think this needs to be update. Right?

Reply:
changed